### PR TITLE
Introduce cri request parsing/generate-hook-request/checkpoing logic

### PR DESCRIPTION
**Pull request by honpey**
### Ⅰ. Describe what this PR does

1. parse cri protocol
2. generate hook request to RuntimeHookServer(koordlet)
3. merge response form RuntimeHookServer to cri request which would be transferred to backend containerd/dockerd
4. checkpoint basing on bolt

### CODE REVIEW

File: pkg/runtime-manager/resource-executor/cri/container.go [hunk 1]
```diff
@@ -0,0 +1,183 @@
+/*
+
```
Comment (jasonliu747) [inline pkg/runtime-manager/resource-executor/cri/container.go:2]: redundant newline

File: pkg/runtime-manager/resource-executor/resource_executor.go [hunk 2]
```diff
@@ -33,6 +41,42 @@ const (
...
+	switch runtimeResourceType {
+	case RuntimePodResource:
+		return cri.NewPodResourceExecutor(store)
+	case RuntimeContainerResource:
+		return cri.NewContainerResourceExecutor(store)
+	}
+	return &NoopResourceExecutor{}
+}
+
+type NoopResourceExecutor struct {
```
[the remark this answers is not in the document]
Comment (honpey) [reply pkg/runtime-manager/resource-executor/resource_executor.go:54]: > What does `Noop` mean? Could you explain briefly?

Noop means no-operation for cri request, where no hook exists like ListContainerStats/ExecSync etc.
comments has beed added.

**Review by jasonliu747 — COMMENTED** [review]
/lgtm

**Pull request merged**